### ISSUE TITLE
OPENVPN disconnecting frequently

### ISSUE THREAD

**Issue by kavi0208** [opnsense/core#3253]
Hello,

Since upgrade from 18.7.10 to 19.1.1, i have several openvpn clients complaining about being disconnected regularly.

In the logs, i can see that openvpn server is restarting after some minutes:
```
Feb 22 13:53:30 gw openvpn[23431]: event_wait : Interrupted system call (code=4)
Feb 22 13:53:30 gw openvpn[23431]: /usr/local/etc/inc/plugins.inc.d/openvpn/ovpn-linkdown ovpns1 1500 1622 10.249.50.1 10.249.50.2 init
Feb 22 13:53:31 gw openvpn[23431]: SIGTERM[hard,] received, process exiting
Feb 22 13:53:31 gw openvpn[24257]: OpenVPN 2.4.6 amd64-portbld-freebsd11.2 [SSL (OpenSSL)] [LZO] [LZ4] [MH/RECVDA] [AEAD] built on Feb  4 2019
Feb 22 13:53:31 gw openvpn[24257]: library versions: OpenSSL 1.0.2q  20 Nov 2018, LZO 2.10
```

I increased the verbosity of openvpn but still was unable to find a reason. I downgraded opnsense from 19.1.1 to 18.7.10 and openvpn works without any problem. 

It seems like openvpn itself does not have any problem as both opnsense uses same version of openvpn:
```
OpenVPN 2.4.6 amd64-portbld-freebsd11.1 [SSL (OpenSSL)] [LZO] [LZ4] [MH/RECVDA] [AEAD] built on Nov 21 2018
```

From what the log states `Feb 22 13:53:30 gw openvpn[23431]: event_wait : Interrupted system call (code=4)` seems like there is a script somewhere which is killing openvpn each time.

Can you help please?

Thanks.

Regards,
Kaviraj

**Comment by AdSchellevis** [comment]
probably network issues, please use our issue templates to report bugs or features.

**Comment by kavi0208** [comment]
**Describe the bug**
Since upgrade from 18.7.10 to 19.1.1, i have several openvpn clients complaining about being disconnected regularly.

In the logs, i can see that openvpn server is restarting after some minutes:

```
Feb 22 13:53:30 gw openvpn[23431]: event_wait : Interrupted system call (code=4)
Feb 22 13:53:30 gw openvpn[23431]: /usr/local/etc/inc/plugins.inc.d/openvpn/ovpn-linkdown ovpns1 1500 1622 10.249.50.1 10.249.50.2 init
Feb 22 13:53:31 gw openvpn[23431]: SIGTERM[hard,] received, process exiting
Feb 22 13:53:31 gw openvpn[24257]: OpenVPN 2.4.6 amd64-portbld-freebsd11.2 [SSL (OpenSSL)] [LZO] [LZ4] [MH/RECVDA] [AEAD] built on Feb  4 2019
Feb 22 13:53:31 gw openvpn[24257]: library versions: OpenSSL 1.0.2q  20 Nov 2018, LZO 2.10
```
I increased the verbosity of openvpn but still was unable to find a reason

**To Reproduce**
Steps to reproduce the behavior:
Upgrade to opnsense 19.1.1

**Expected behavior**
No more disconnection

**Screenshots**

**Relevant log files**
```
Feb 22 13:53:30 gw openvpn[23431]: event_wait : Interrupted system call (code=4)
Feb 22 13:53:30 gw openvpn[23431]: /usr/local/etc/inc/plugins.inc.d/openvpn/ovpn-linkdown ovpns1 1500 1622 10.249.50.1 10.249.50.2 init
Feb 22 13:53:31 gw openvpn[23431]: SIGTERM[hard,] received, process exiting
Feb 22 13:53:31 gw openvpn[24257]: OpenVPN 2.4.6 amd64-portbld-freebsd11.2 [SSL (OpenSSL)] [LZO] [LZ4] [MH/RECVDA] [AEAD] built on Feb  4 2019
Feb 22 13:53:31 gw openvpn[24257]: library versions: OpenSSL 1.0.2q  20 Nov 2018, LZO 2.10
```

**Environment**
```
FreeBSD gw 11.1-RELEASE-p18 FreeBSD 11.1-RELEASE-p18  4b57161bcff(stable/18.7)  amd64
OpenVPN 2.4.6 amd64-portbld-freebsd11.1 [SSL (OpenSSL)] [LZO] [LZ4] [MH/RECVDA] [AEAD] built on Nov 21 2018
```

OPNsense 19.1.1 (amd64, OpenSSL).  
Virtual machine.

The problem is not present on opnsense version 18.7.10

**Comment by fichtner** [comment]
DHCP on WAN? How regularly are you disconnected... like clockwork every X minutes?

**Comment by kavi0208** [comment]
Fixed IP on wan. For the disconnection it is not regular the minimum we observed is 3min after being connected and the max is about 15min.

**Comment by fichtner** [comment]
> FreeBSD gw 11.1-RELEASE-p18 FreeBSD 11.1-RELEASE-p18  4b57161bcff(stable/18.7)  amd64

Not the correct kernel as this is for 18.7, can you do a full in-place reinstall:

# opnsense-update -f

and reboot? no settings will be lost...

**Comment by fichtner** [comment]
Sorry unless you mean that is the working version...

**Comment by kavi0208** [comment]
Yea it is working fine with 18.7. Problem is only with 19.1.1. (i had to rollback to stabilise)

**Comment by kavi0208** [comment]
@fichtner The problem is solved with 19.1.2.

Closing this issue.

Thanks.

**Comment by kavi0208** [comment]
changelog 19.1.2 openvpn: proper daemonize instead of background job
